### PR TITLE
Upgrade NDK to v15c.

### PR DESCRIPTION
**Pull request by jdm**
This upgrades us to a more recent NDK to address that outstanding technical debt.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/21780)

### CODE REVIEW

**Comment by atouchet** [comment]
If APP_PLATFORM is being updated to 21 should minSdkVersion in https://github.com/servo/servo/blob/master/support/android/apk/servoapp/build.gradle be set to 21 as well?

**Comment by jdm** [comment]
This is a prerequisite for the android build in #23163.

**Review by asajeffrey — APPROVED** [review]
LGTM. Oh the joys of guessing paths in the NDK.

File: python/servo/build_commands.py [hunk 1]
```diff
@@ -339,15 +339,15 @@ def build(self, target=None, release=False, dev=False, jobs=None,
...
             if not os.path.isfile(path.join(env["ANDROID_NDK"], 'source.properties')):
                 print("ANDROID_NDK should have file `source.properties`.")
                 print("The environment variable ANDROID_NDK may be set at a wrong path.")
                 sys.exit(1)
             with open(path.join(env["ANDROID_NDK"], 'source.properties')) as ndk_properties:
                 lines = ndk_properties.readlines()
-                if lines[1].split(' = ')[1].split('.')[0] != '12':
-                    print("Currently only support NDK 12.")
+                if lines[1].split(' = ')[1].split('.')[0] != '15':
+                    print("Currently only support NDK 15.")
```
Comment (Manishearth) [inline python/servo/build_commands.py]: Probably should also say something like "please delete the android-toolchains folder and rerun bootstrap-android"

**Comment by jdm** [comment]
Verified that this produces a runnable android build on my pixel 1.
@bors-servo r=asajeffrey

**Pull request merged**